experimental/coinbase-negotiator/README.md

```diff
@@ -0,0 +1,168 @@
+          proposed
+bit  a+1: understand Schnorr + presigned txs negotiation
+bit  a+2: understand CTV negotiation 
+```
+
+### Schnorr + presigned txs negotiation:
+1. Each client send to the pool a public key and the address where it want to retrieve the coins
+2. The pool create coinbase that sent the input to an address (a1) obtained from the "sum" of all
+   the client's pub key
+3. The pool create a transaction for each client. The created tx send an amount from a1 to the address
+   provided by the client in step 1
+4. Each client sign each transaction created in step 3
+5. coinbase from step 2 is used in the next block
+
+In order to calculate the amount that the presigned transaction must transfer from a1 to the client
+address in step 2, the pool just use the already provided hashrate by the client that has not been
+paid yet (so what the miner is being paid is not the block that contribute to mine but the previous
+one/ones)
+
+#### Pros:
+1. pool is noncustodial
+2. miners can verify the 2 below prop of the coinbase tx that they are mining (a1 is the output of the coinbase tx)
+    1. they can see that a valid transaction from a1 to the miner address exist and have the *right*
+       amount
+    2. they need to sign each valid tx from a1 so they can see that the total amount of all the
+       presigned txs from a1 is (coinbase input - the fair reward that they expect)
+
+#### Cons:
+1. not scale  well on chain, cause if you are a small miner and you get paid at each found
+   blocks you will likely end up paying a lot in fees. You can mitigate it (1) sending the txs when the
+   fee is small, (2) aggregating txs, (3) the pool is not paying you at each founded block but only
+   when you will be paid an amount above a minimum one.
+2. not scale well off chain: pool need to coordinate all the miners, and everyone need to agree and be
+   online if only one fail the process need to be restarted from zero.
+
+### CTV
+Using CTV you will have the above pros without the above cons so it make sense to add this
+possibility if it will be merged on master.
+
+### No coinbase negotiation
+Pool can reserve a path to itself where they do custody bitcoin for miner that prefer to use a custodian pool.
+
+### Messages
+
+#### Valid for both Schnorr and CTV negotiations:
+
+Upstream will use the provided address as receiving address for the downstream path.
+```
+Client -> Server
+NewAddress:
+    address: B0255
+    
+msg_type: 0x77
+```
+
+#### Valid for Schnorr negotiations:
+
+
+Upstream will use the pub keys as Xi and Ri to calculate X and R as described here
+https://blog.blockstream.com/en-musig-key-aggregation-schnorr-signatures/ (MuSig paragrapher)
+```
+Client -> Server
+NewPubKeyPair:
+    pub_key_x: B0255
+    pub_key_r: B0255
+
+msg_type: 0x78
+```
+
+
+
+Downstream will use X, R, m, and L to sign m as described here
+https://blog.blockstream.com/en-musig-key-aggregation-schnorr-signatures/ (MuSig paragrapher)
```

Comment:
Downstream will use `X`, `R`, `m`, and `L` to sign m as described [here](https://blog.blockstream.com/en-musig-key-aggregation-schnorr-signatures/).

(MuSig paragrapher) -> Is this a TODO?